pkg/apis/config/v1alpha1/metrics/config.go

```diff
@@ -14,21 +14,15 @@
 
 package metrics
 
-import (
-	model "github.com/prometheus/client_model/go"
-)
-
-// Gather is our prometheus.Gatherer interface for proxying metrics.
-func (m *Metrics) Gather() ([]*model.MetricFamily, error) {
-	m.Lock()
-	pend := m.pend
-	m.Unlock()
-
-	if pend == nil {
-		log.Debug("no data to proxy to prometheus...")
-	} else {
-		log.Debug("proxying data to prometheus...")
-	}
-
-	return pend, nil
+// Config provides runtime configuration for metrics collection.
+// +k8s:deepcopy-gen=true
```

Comment:
please drop this line, it is not used anymore. See the migration guide for the reference: https://github.com/kubernetes-sigs/kubebuilder/blob/782a4da58456003c7da3c3ccdac6558328218de8/docs/book/src/migration/legacy/migration_guide_v1tov2.md

Reply:
Okay, so that talks about kubebuilder v2. What is the relation of that to `controller-tools v0.16.5` and `code-generator v0.28.1`. Are those v2 or v1 ? We are non-urgently a bit stuck with those versions ATM, until we figure out what is the equivalent of this
```shell
code-generator/generate-groups.sh client \
            github.com/containers/nri-plugins/pkg/generated \
            github.com/containers/nri-plugins/pkg/apis \
            "config:v1alpha1" --output-base=. \
            --go-header-file=./docs/license-header.go.txt && \
        rm -fr pkg/generated && \
        mv github.com/containers/nri-plugins/pkg/generated pkg && \
        rm -fr github.com
```
in the new version, which is neither a drop-in replacement nor offers a backward-compatibility wrapper script.

That is why we have our versions locked to the above mentioned ones, because nothing forces us to bump those versions but the lack of backward-compatibility prevents us until we figure out how to generate all we need (types, manifests/CRDs, clientsets) with the new toolset. The last thing forcing a version update was the lack of CEL-expression based validation, but bumping to a more but not too recent version solved that.

Reply:
Ok, I will take care of that then. You can ignore my comment here.